packages/env-provider/src/services/index.ts

```diff
@@ -0,0 +1,107 @@
+/*-
+ *
+ * Hedera JSON RPC Relay
+ *
+ * Copyright (C) 2022-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import { IEnvProviderService } from './IEnvProviderService';
+import dotenv from 'dotenv';
+import findConfig from 'find-config';
+
+export class EnvProviderService implements IEnvProviderService {
+  /**
+   * The singleton instance
+   * @public
+   */
+  private static instance: IEnvProviderService;
+
+  /**
+   * Copied envs from process.env
+   * @private
+   */
+  private envs: JSON;
+
+  /**
+   * Fetches all envs from process.env and pushes them into the envs property
+   * @private
+   */
+  private constructor() {
+    dotenv.config({ path: findConfig('.env') || '' });
+    this.envs = JSON.parse(JSON.stringify(process.env));
+  }
+
+  /**
+   * Get the singleton instance of the current service
+   * @public
+   */
+  public static getInstance(): IEnvProviderService {
+    if (this.instance == null) {
+      this.instance = new EnvProviderService();
+    }
+
+    return this.instance;
+  }
+
+  /**
+   * Hot reload a new instance into the current one, used in test cases only
+   */
+  public static hotReload(): void {
+    this.instance = new EnvProviderService();
+  }
+
+  /**
+   * Get an env var by name
+   * @param name string
+   * @returns string | undefined
+   */
+  public get(name: string): string | undefined {
```

Comment:
We can implement a default value or fallback mechanism to handle missing environment variables:

```typescript
public getWithDefault(name: string, defaultValue: string): string {
  const value = this.envs[name];
  if (!value) {
    console.warn(`Environment variable ${name} is not set, using default: ${defaultValue}`);
    return defaultValue;
  }
  return value;
}
```

We can also throw an error or log a warning when a required environment variable is not set:

```typescript
public getRequired(name: string): string {
  const value = this.envs[name];
  if (!value) {
    throw new Error(`Missing required environment variable: ${name}`);
  }
  return value;
}
```